src/ChannelAdvisorAccess/REST/Services/ServiceBaseAbstr.cs

```
@@ -625,23 +625,34 @@ private void AddAccountInfoToHeader()
 							string content = await httpResponse.Content.ReadAsStringAsync().ConfigureAwait( false );
 							RefreshLastNetworkActivityTime();
 
-							int batchStatusCode;
-							IEnumerable< T > parsedEntities;
-							if ( MultiPartResponseParser.TryParse< T >( content, out batchStatusCode, out parsedEntities ) )
-							{
-								entities.AddRange( parsedEntities );
+							int batchStatusCode = ( int )HttpStatusCode.OK;
+
+							try
+							{								
+								IEnumerable< T > parsedEntities;
+								if ( MultiPartResponseParser.TryParse< T >( content, out batchStatusCode, out parsedEntities ) )
+								{
+									entities.AddRange( parsedEntities );
+								}
+								else
+								{
+									var error = String.Format( "Can't parse the response: {0}", content );
+									ChannelAdvisorLogger.LogTrace( error );
+									throw new ChannelAdvisorException( (int) HttpStatusCode.InternalServerError, error );

```

@maxim-saltanov-skuvault , Are you sure this needs to throw since in Prod we're just setting `batchStatusCode = (int)httpResponse.StatusCode;` in this "else" and then advancing to the next batch? We could still add logging but I didn't think we wanted to change how the app functions at this time.

---

@maxim-skuvault I think that if we get a response that we cannot parse, then we should throw an error. If we do not do this, then as you said earlier, we will simply skip some of the products and synchronize the rest and the inventorysync integration will be displayed as successful - I think this is wrong. Probably we will need to discuss it with BA...